### PR TITLE
Migrate Configuration Discovery Tests from Groovy to Java

### PR DESCRIPTION
**Pull request by wborn**
Migrates the remaining Configuration Discovery Tests from Groovy to Java.

### CODE REVIEW

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java [hunk 1]
```diff
@@ -673,29 +674,29 @@ protected void unsetEventPublisher(EventPublisher eventPublisher) {
     }
 
     @Reference
-    protected void setThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
+    public void setThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
```
Comment (htreu) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java]: why is this change necessary (and the other protected to public changes, too)?

Comment (wborn) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java]: I think I changed them when Eclipse reported visibility errors in PersistentInboxTest. For some reason they are gone now. I've also moved the InboxOSGITest to another package so it can also use the protected methods.

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java [hunk 2]
```diff
@@ -97,9 +97,9 @@
      * Internal enumeration to identify the correct type of the event to be fired.
      */
     private enum EventType {
-        added,
-        removed,
-        updated
+    added,
+    removed,
+    updated
```
Comment (wborn) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java]: This change was done by the automatic formatting. :cry:

Comment (htreu) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java]: yes, very strange behaviour. Reported here: https://bugs.eclipse.org/bugs/show_bug.cgi?id=536322
Please try the workaround and reduce the JavaDoc to a simple comment like `// Internal ....`.

Comment (wborn) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java]: Thanks, the workaround seems to work. :+1:
